sale_coupon_multi_use/models/sale_coupon.py

```diff
@@ -63,32 +55,24 @@ def _get_related_sale_order_line(self, sale_order):
             lambda r: r.product_id == discount_product
         )[0]
 
-    def _prepare_consumption_line(self, amount_consumed, sale_order):
+    def _prepare_consumption_line(self, sale_order_line):
         self.ensure_one()
-        line = self._get_related_sale_order_line(sale_order)
         return {
             "coupon_id": self.id,
-            "amount": amount_consumed,
-            "sale_order_line_id": line.id,
+            "amount": abs(sale_order_line.price_total),
```

Comment:
do we need to check if the amount of coupon is not exceeded before line creation 
like if the amount on coupon left is 20 and order_line.price_total is 50
IMO this should be filtered earlier than on creation but I see filtering on the program not on the coupon, maybe I missing smth

Reply:
The way it works now, it finds current discount delta (remaining amount for discount). Then it compares with existing sale order total amount. And it uses the lower amount -> so it won't apply higher discount than order itself and it won't apply higher discount than there is discount delta (remaining amount).

And that line created is actually used to finalize consumption and return flag whether coupon is fully consumed. In previous implementation, it worked differently, where it would check what kind of amount to use for consumption line, but it was actually more problematic, so I redesigned to intercept earlier (when sale_coupon module actually tries to add discount on sale order line). This have multiple advantages. You can read commit message for that.